includes/class-wc-stripe-webhook-handler.php

```diff
@@ -106,36 +106,49 @@ public function validate_request( $request_headers, $request_body ) {
-			return WC_Stripe_Webhook_State::VALIDATION_FAILED_USER_AGENT_INVALID;
+		if ( empty( $this->secret ) ) {
+			return $this->validate_request_user_agent( $request_headers );
 		}
 
-		if ( ! empty( $this->secret ) ) {
-			// Check for a valid signature.
-			$signature_format = '/^t=(?P<timestamp>\d+)(?P<signatures>(,v\d+=[a-z0-9]+){1,2})$/';
-			if ( empty( $request_headers['STRIPE-SIGNATURE'] ) || ! preg_match( $signature_format, $request_headers['STRIPE-SIGNATURE'], $matches ) ) {
-				return WC_Stripe_Webhook_State::VALIDATION_FAILED_SIGNATURE_INVALID;
-			}
+		// Check for a valid signature.
+		$signature_format = '/^t=(?P<timestamp>\d+)(?P<signatures>(,v\d+=[a-z0-9]+){1,2})$/';
+		if ( empty( $request_headers['STRIPE-SIGNATURE'] ) || ! preg_match( $signature_format, $request_headers['STRIPE-SIGNATURE'], $matches ) ) {
+			return WC_Stripe_Webhook_State::VALIDATION_FAILED_SIGNATURE_INVALID;
+		}
 
-			// Verify the timestamp.
-			$timestamp = intval( $matches['timestamp'] );
-			if ( abs( $timestamp - time() ) > 5 * MINUTE_IN_SECONDS ) {
-				return WC_Stripe_Webhook_State::VALIDATION_FAILED_TIMESTAMP_MISMATCH;
-			}
+		// Verify the timestamp.
+		$timestamp = intval( $matches['timestamp'] );
+		if ( abs( $timestamp - time() ) > 5 * MINUTE_IN_SECONDS ) {
+			return WC_Stripe_Webhook_State::VALIDATION_FAILED_TIMESTAMP_MISMATCH;
+		}
 
-			// Generate the expected signature.
-			$signed_payload     = $timestamp . '.' . $request_body;
-			$expected_signature = hash_hmac( 'sha256', $signed_payload, $this->secret );
+		// Generate the expected signature.
+		$signed_payload     = $timestamp . '.' . $request_body;
+		$expected_signature = hash_hmac( 'sha256', $signed_payload, $this->secret );
 
-			// Check if the expected signature is present.
-			if ( ! preg_match( '/,v\d+=' . preg_quote( $expected_signature, '/' ) . '/', $matches['signatures'] ) ) {
-				return WC_Stripe_Webhook_State::VALIDATION_FAILED_SIGNATURE_MISMATCH;
-			}
+		// Check if the expected signature is present.
+		if ( ! preg_match( '/,v\d+=' . preg_quote( $expected_signature, '/' ) . '/', $matches['signatures'] ) ) {
+			return WC_Stripe_Webhook_State::VALIDATION_FAILED_SIGNATURE_MISMATCH;
 		}
 
 		return WC_Stripe_Webhook_State::VALIDATION_SUCCEEDED;
 	}
 
+	/**
+	 * Verify User Agent of the incoming webhook notification. Used as fallback for the cases when webhook secret is missing.
+	 *
+	 * @since 5.0.0
+	 * @version 5.0.0
+	 * @param array $request_headers The request headers from Stripe.
+	 * @return string The validation result (e.g. self::VALIDATION_SUCCEEDED )
+	 */
+	private function validate_request_user_agent( $request_headers ) {
+		$ua_is_valid = empty( $request_headers['USER-AGENT'] ) || preg_match( '/Stripe/', $request_headers['USER-AGENT'] );
+		$ua_is_valid = apply_filters( 'wc_stripe_webhook_validate_user_agent', $ua_is_valid, $request_headers );
```

Comment:
Before looking at the code, I was confused about what was expected from this filter because of its name.

I didn't know if it should validate the User-Agent or if it would perform User-Agent validation.

Perhaps a clearer description of what it does could be something like `wc_stripe_webhook_is_valid_user_agent`. - Thinking in terms of function naming. 

But I'm still new to WP, so if that kind of nomenclature doesn't apply to filters/actions, you can ignore this comment.

Reply:
Good point, renamed in https://github.com/woocommerce/woocommerce-gateway-stripe/pull/1493/commits/19f18350844c2c4a2615f9cc8f9f81bb176575d2.